docs/design/proposals/late-initialization/late-initialization.md

```diff
@@ -0,0 +1,277 @@
+the K8s resource in the “etcd” does not correctly represent the actual state of resource in AWS. This is a problem because
+for K8s user, the spec of custom resource is the source of truth for state of K8s resource.
+
+b) When a default field is populated during AWS resource creation, then during reconciliation, there will always be delta
+present between desired and latest K8s representation of the custom resource and an Update AWS API call be made during
+reconciliation. This update call may or may not have any side-effects on AWS resource and currently service teams either
+place custom code to avoid this OR let the calls happen if it does not alter state of the resource.
+
+> Problem (b) can be viewed as a variation of Problem (a), where server changes the property value from “null” to something.
+This is worth noting because solving Problem (a) in a generic and consistent manner will provide the solution for
+Problem (b) as well.
+
+
+## Solutions
+
+To compare the solutions for above problem, we will use the following facets:
+
+1. Consistency with K8s conventions
+2. Consistency across ACK service controllers
+3. GitOps compatibility
+4. User experience for custom resource lifecycle 
+
+✅ → doable with no/some efforts
+
+❌ → not achievable
+
+### Solution 1
+
+In this solution, the status fields will include all the fields which are CreateOutput shape members. Current status 
+fields are the CreateOutput shape members which are not present in CreateInput shape (spec fields).
+
+During reconciliation, the Read(One|Many),Update request inputs will be created from desired state but giving precedence
+to status fields over spec in case of conflict.
+
+
+1. *Consistency with K8s conventions* ❌ 
+    1. In K8s convention, the spec fields are not duplicated into status fields. This will confuse k8s users when they 
+    describe their custom resources.
+2. *Consistency across ACK service controllers* ✅
+    1. There should be no complexity in following this approach across all ACK service controllers.
+3. *GitOps compatibility* ✅
+    1. There should be no complexity in implementing GitOps for ACK custom resources. 
+    2. Approach mentioned in low level design below should work.
+4. *User experience for custom resource lifecycle* ✅
+    1. Applying updates to custom resource can become confusing for the user.
+    2. To update custom resource spec, K8s user will have to check resource status because status will have most updated
+     value of all the fields.
+
+Other Cons :
+
+* The number of status fields will be very large and there will be a lot of duplication in custom resource description
+
+
+### Solution 2
+
+This solution builds on top of “Solution 1”, with the assumption that “Problem” mentioned above is not very common 
+amongst AWS APIs.
+
+In this solution, service teams can specify the spec fields which will be duplicated in both spec and status of custom
+resource using generator.yaml
```

Comment:
Unfortunately I don't think this provides any value over solution 1, except for reducing duplication. Understand the reasoning for proposing this, though.